platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheProblems.kt

```diff
@@ -152,10 +161,54 @@ class ConfigurationCacheProblems(
     private
     fun onProblem(problem: PropertyProblem, severity: ProblemSeverity) {
         if (summarizer.onProblem(problem, severity)) {
+            problemsService.onProblem(problem, severity)
             report.onProblem(problem)
         }
     }
 
+    private
+    fun InternalProblems.onProblem(problem: PropertyProblem, severity: ProblemSeverity) {
+        val message = problem.message.toString()
+        internalReporter.create {
+            id(
+                "configuration-cache-" + DeprecationMessageBuilder.createDefaultDeprecationId(message),
+                message,
+                GradleCoreProblemGroup.validation()
+            ) // TODO (reinhold) display name might be too elaborate
```

Comment:
Please double-check that. The id should be just a few words. We have the `contextualLabel` and the `details` fields to add longer descriptions.